src/TreeView/TreeView.tsx

```diff
@@ -0,0 +1,217 @@
+import {ChevronDownIcon, ChevronRightIcon} from '@primer/octicons-react'
+import React from 'react'
+import Box from '../Box'
+
+// ----------------------------------------------------------------------------
+// Context
+
+const ItemContext = React.createContext<{
+  level: number
+  isExpanded: boolean
+}>({
+  level: 1,
+  isExpanded: false
+})
+
+// ----------------------------------------------------------------------------
+// TreeView
+
+export type TreeViewProps = {
+  'aria-label'?: React.AriaAttributes['aria-label']
+  'aria-labelledby'?: React.AriaAttributes['aria-labelledby']
+  children: React.ReactNode
+}
+
+const Root: React.FC<TreeViewProps> = ({'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledby, children}) => {
+  return (
+    <Box
+      as="ul"
+      role="tree"
+      aria-label={ariaLabel}
+      aria-labelledby={ariaLabelledby}
+      sx={{
+        listStyle: 'none',
+        padding: 0,
+        margin: 0
+      }}
+    >
+      {children}
+    </Box>
+  )
+}
+
+// ----------------------------------------------------------------------------
+// TreeView.Item
+
+export type TreeViewItemProps = {
+  children: React.ReactNode
+  onSelect?: (event: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) => void
+  onToggle?: (isExpanded: boolean) => void
+}
+
+const Item: React.FC<TreeViewItemProps> = ({onSelect, onToggle, children}) => {
+  const {level} = React.useContext(ItemContext)
+  const [isExpanded, setIsExpanded] = React.useState(false)
+  const {hasSubTree, subTree, childrenWithoutSubTree} = useSubTree(children)
+
+  // Expand or collapse the subtree
+  const toggle = React.useCallback(() => {
+    onToggle?.(!isExpanded)
+    setIsExpanded(!isExpanded)
+  }, [isExpanded, onToggle])
```

Comment:
This seems like something that would be re-evaluated each `render` if `onToggle` does not have a stable identity (e.g. it's passed in as an inline arrow function expression). If that's the case, what would you think about making this an inline arrow function or function declaration?

Reply:
Great catch! I'll do that